src/util.ts

```diff
@@ -17,6 +18,16 @@ export function hash(a: any) {
   return stringify(a);
 }
 
+export function empty(a: any) {
```

Comment:
Do we really need this mixed up utility?  

Looking at where you use this method, your `a` is always an object.  

Why don't you just call `keys(a).length === 0` just like everywhere else in the codebase?